File: csv_summary_stats.R
# R Script to Load CSV and Print Summary Statistics
# Generate random filename for CSV loading

# Set seed for reproducibility
set.seed(123)

# Generate a random filename
random_filename <- paste0("data_", sample(1000:9999, 1), ".csv")

# Check if file exists, if not create a sample dataset and save it
if (!file.exists(random_filename)) {
  # Create sample data
  sample_data <- data.frame(
    ID = 1:100,
    Age = sample(18:80, 100, replace = TRUE),
    Income = round(rnorm(100, 50000, 15000)),
    Score = runif(100, 0, 100)
  )
  
  # Write to CSV
  write.csv(sample_data, random_filename, row.names = FALSE)
  cat("Created sample data file:", random_filename, "\n")
}

# Load the CSV file
data <- read.csv(random_filename)

# Print summary statistics
cat("Summary Statistics for", random_filename, "\n")
cat("=====================================\n")
print(summary(data))

# Additional statistics
cat("\nStandard Deviations:\n")
print(sapply(data[sapply(data, is.numeric)], sd))

cat("\nData Structure:\n")
str(data)

File: csv_summary_stats.py
import pandas as pd
import numpy as np
import random
import os

# Set seed for reproducibility
random.seed(123)
np.random.seed(123)

# Generate a random filename
random_filename = f"data_{random.randint(1000, 9999)}.csv"

# Check if file exists, if not create a sample dataset and save it
if not os.path.exists(random_filename):
    # Create sample data
    sample_data = pd.DataFrame({
        'ID': range(1, 101),
        'Age': np.random.randint(18, 81, 100),
        'Income': np.round(np.random.normal(50000, 15000, 100)).astype(int),
        'Score': np.random.uniform(0, 100, 100)
    })
    
    # Write to CSV
    sample_data.to_csv(random_filename, index=False)
    print(f"Created sample data file: {random_filename}")

# Load the CSV file
data = pd.read_csv(random_filename)

# Print summary statistics
print(f"Summary Statistics for {random_filename}")
print("=" * (30 + len(random_filename)))
print(data.describe())

# Additional statistics
print("\nData Info:")
print(data.info())